src/v/debug_bundle/debug_bundle_service.cc

```diff
@@ -70,6 +73,20 @@ std::filesystem::path form_debug_bundle_file_path(
   const std::filesystem::path& base_path, job_id_t job_id) {
     return base_path / fmt::format("{}.zip", job_id);
 }
+
+bool is_valid_rfc1123(std::string_view ns) {
+    // Regular expression for RFC1123 hostname validation
+    constexpr std::string_view rfc1123_pattern
+      = R"(^([a-zA-Z0-9]([a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?))";
+
+    // Validate the hostname against the regular expression using RE2
+    return RE2::FullMatch(ns, RE2(rfc1123_pattern));
+}
+
+bool is_valid_k8s_namespace(std::string_view ns) {
+    constexpr auto max_ns_length = 63;
+    return !ns.empty() && ns.size() <= max_ns_length && is_valid_rfc1123(ns);
+}
```

Comment:
nitpick: Since the RE2 isn't compiled (and stored in a static const, or something), it makes sense to do the length checks, otherwise they are redundant.

```suggestion
bool is_valid_k8s_namespace(std::string_view ns) {
    // Regular expression for RFC1123 hostname validation
    static const RE2 rfc1123_pattern{
      R"(^([a-zA-Z0-9]([a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?))"};

    // Validate the hostname against the regular expression using RE2
    return RE2::FullMatch(ns, rfc1123_pattern);
}
```

Reply:
Actually, is this right? Do we need to support single character namespaces?